### DISCUSSION TITLE
Basic sample using Jetpack Compose

### DISCUSSION THREAD

**Discussion by sarimmehdi** [sceneview/sceneview#223] [General]
It's great the project now supports compose but a simple app showing how to use it with compose would be really helpful

**Comment by Blizl** [comment]
I have a PR open for this https://github.com/SceneView/sceneview-android/pull/300

**Reply by sadrabt** [reply]
I found your sample very useful so just wanted to say thank you

**Comment by ThomasGorisse** [comment]
SceneView v4.0.0-rc.1 is fully Compose-native. Basic 3D viewer:

```kotlin
implementation("io.github.sceneview:sceneview:4.0.0-rc.1")

@Composable
fun ModelViewer() {
    SceneView {
        val model = rememberModelInstance(modelLoader, "models/damaged_helmet.glb")
        if (model != null) {
            ModelNode(modelInstance = model)
        }
    }
}
```

Check `samples/android-demo` for a full Material 3 app with 19 sample demos covering model loading, animations, geometry nodes, image/video nodes, and more.

**Comment by ThomasGorisse** [comment]
SceneView v4 is fully Compose-native. Here's a complete minimal example:

```kotlin
@Composable
fun My3DViewer() {
    val engine = rememberEngine()
    val modelLoader = rememberModelLoader(engine)
    val model = rememberModelInstance(modelLoader, "models/damaged_helmet.glb")

    SceneView(
        modifier = Modifier.fillMaxSize(),
        engine = engine,
        modelLoader = modelLoader
    ) {
        if (model != null) {
            ModelNode(
                modelInstance = model,
                scaleToUnits = 1f
            )
        }
    }
}
```

Add the dependency:
```kotlin
implementation("io.github.sceneview:sceneview:4.0.0-rc.1")
```

For AR:
```kotlin
implementation("io.github.sceneview:arsceneview:4.0.0-rc.1")
```

See the [demo app](https://github.com/sceneview/sceneview/tree/main/samples/android-demo) for 19 complete examples.
